File: utils/torCheck.js
const axios = require('axios');

async function checkTorRunning() {
  try {
    await axios.get('http://check.torproject.org', { proxy: { host: '127.0.0.1', port: 9050, protocol: 'socks5h' } });
  } catch (error) {
    console.error('Error: Tor is not running or not properly configured. Please start Tor and try again.');
    process.exit(1);
  }
}

async function checkOnionSite(url) {
  if (!url.startsWith('http://') && !url.startsWith('https://')) {
    url = 'http://' + url;
  }

  try {
    const response = await axios.get(url, { proxy: { host: '127.0.0.1', port: 9050, protocol: 'socks5h' } });
    if (response.status === 200) {
      return { siteExists: true, message: 'Site exists and is reachable.' };
    } else {
      return { siteExists: false, message: `Site returned status code: ${response.status}` };
    }
  } catch (error) {
    return { siteExists: false, message: `Error fetching the page: ${error.message}` };
  }
}

module.exports = { checkTorRunning, checkOnionSite };
      

File: torspy/scraper/tor_check.py
#!/usr/bin/python3
# Copyright (©️) 2024 author: Fidal
# Issue : https://github.com/mr-fidal/torspy

import requests
import sys

def check_tor_running():
    try:
        session = requests.session()
        session.proxies = {
            'http': 'socks5h://127.0.0.1:9050',
            'https': 'socks5h://127.0.0.1:9050'
        }
        response = session.get('http://check.torproject.org', timeout=10)
        if "Congratulations. This browser is configured to use Tor." not in response.text:
            raise Exception("Tor is not properly configured.")
    except Exception as e:
        print("Error : Tor is not running or not properly configured. Please start Tor and try again.")
        sys.exit(1)
